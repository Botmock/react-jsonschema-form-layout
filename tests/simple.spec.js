import React from 'react'
import LayoutField from '../src'
import renderer from 'react-test-renderer'
import Form from 'react-jsonschema-form'

const schema = {
  title: 'Todo',
  type: 'object',
  required: ['title'],
  properties: {
    'password': {
      'type': 'string',
      'title': 'Password'
    },
    'lastName': {
      'type': 'string',
      'title': 'Last name'
    },
    'bio': {
      'type': 'string',
      'title': 'Bio'
    },
    'firstName': {
      'type': 'string',
      'title': 'First name'
    },
    'age': {
      'type': 'integer',
      'title': 'Age'
    }
  }
}

const fields = {
  layout: LayoutField
}

const uiSchema = {
  'ui:field': 'layout',
  'ui:options': {
    layout: [
      {
        firstName: { md: 6 },
        lastName: { md: 6 }
      }, {
        bio: { md: 12 }
      }, {
        age: { md: 6 },
        password: { md: 6 }
      }
    ]
  }
}


test('Layout should render correctly', () => {
  const component = renderer.create(
    <Form schema={schema} uiSchema={uiSchema} fields={fields} />
  )
  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()

})
