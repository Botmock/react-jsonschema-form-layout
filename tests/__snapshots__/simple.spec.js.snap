// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render correctly 1`] = `
<div>
  <h2>
    Welcome to IMPLEMENATION cooleact components
  </h2>
</div>
`;
